src/Raven.Server/Documents/Handlers/AttachmentHandler.cs

```diff
@@ -387,6 +387,183 @@ public async Task Delete()
             }
         }
 
+        [RavenAction("/databases/*/debug/attachments/missing", "GET", AuthorizationStatus.ValidUser, EndpointType.Read, DisableOnCpuCreditsExhaustion = true)]
+        public async Task GetMissingAttachments()
+        {
+            var types = new List<AttachmentType>(2);
+            var typeString = GetStringQueryString("type", false);
+            var collection = GetStringQueryString("collection", true);
+            if (string.IsNullOrEmpty(typeString) == false)
+            {
+                if (Enum.TryParse(typeString, true, out AttachmentType type) == false)
+                {
+                    throw new ArgumentException($"Query string '{typeString}' was not recognized as valid type");
+                }
+
+                types.Add(type);
+            }
+            else
+            {
+                types.Add(AttachmentType.Document);
+                types.Add(AttachmentType.Revision);
+            }
+
+            if (string.IsNullOrEmpty(collection) == false)
+            {
+                if (collection.Equals(Constants.Documents.Collections.AllDocumentsCollection, StringComparison.OrdinalIgnoreCase) == false)
+                {
+                    if (Database.DocumentsStorage.GetCollection(collection, true) == null)
+                        throw new ArgumentException($"Query string '{collection}' was not recognized as valid collection name");
+                }
+            }
+
+            var start = GetLongQueryString("start", false) ?? 0;
+            var pageSize = GetPageSize();
+
+            using (var token = CreateHttpRequestBoundOperationToken())
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (context.OpenReadTransaction())
+            await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                writer.WriteStartObject();
+
+                var deserializationRoutine = JsonDeserializationBase.GenerateJsonDeserializationRoutine<MissingAttachmentInfo>();
```

Comment:
why this passed as param if we have only single one?

Reply:
done, leftover of refactoring